finesse/hardware/plugins/decades/decades.py

```diff
@@ -120,9 +65,91 @@ def __init__(
             poll_interval: How often to poll the sensors (seconds)
         """
         super().__init__(DECADES_URL.format(host=host))
+        self._poll_interval = poll_interval
         self._poll_timer = QTimer()
         self._poll_timer.timeout.connect(self.send_data)
-        self._poll_timer.start(int(poll_interval * 1000))
+        self._poll_timer.start(int(self._poll_interval * 1000))
+
+        # Obtain full parameter list in order to parse received data
+        self.send_params()
+
+        # Poll device once on open.
+        # TODO: Run this synchronously so we can check that things work before the
+        # device.opened message is sent
+        self.send_data()
+
+    def send_params(self) -> None:
```

Comment:
Shouldn't this be called `get_params" given that we are requesting the parameters form the server?

Reply:
I called this `send_params` to be consistent with `send_data`, which is the name we use elsewhere for other devices. I guess we're requesting the device to send the data/params. I agree that it sounds weird though. Maybe we should rename across devices.

Reply:
OK, it's just the EM27 sensor module which also defines `send_data`. Perhaps in fixing #457 we can address the naming issue there. Tagging @alexdewar

Reply:
Yeah, I renamed it to `request_readings()` in my generic `SensorsBase` class. Let's just leave it as is for now and I'll make the naming consistent when I port it (#567)

Reply:
Makes sense! I should have checked your PR first haha